plugins/resources/unixfilesystem/src/unixfilesystem.cpp

```diff
@@ -36,153 +42,253 @@
+auto unix_file_start_operation(irods::plugin_property_map& prop_map) -> irods::error
+{
+    using logger = irods::experimental::log;
+    irods::error ret = SUCCESS();
+    bool detached_mode = is_operating_in_detached_mode(prop_map);
+
+    if (detached_mode) {
+        bool in_host_list = false;
+        std::optional<std::string> new_vault_path_optional;
+
+        // update host to new host
+        char local_hostname[MAX_NAME_LEN];
+        gethostname(local_hostname, MAX_NAME_LEN);
+
+        auto resource_hostname_optional = resolve_hostname(local_hostname, hostname_resolution_scheme::match_preferred);
+
+        std::string resource_hostname;
+        resource_hostname = resource_hostname_optional ? std::move(*resource_hostname_optional) : local_hostname;
+
+        std::string resource_name;
+        ret = prop_map.get<std::string>(irods::RESOURCE_NAME, resource_name);
+        if (!ret.ok()) {
+            logger::resource::warn("[resource_name={}] Detached mode failed to set RESOURCE_HOST to {}."
+                                   "  Failed to get irods::RESOURCE_NAME property.",
+                                   resource_name.c_str(),
+                                   local_hostname);
+            return SUCCESS();
+        }
+
+        std::tie(in_host_list, new_vault_path_optional) =
+            get_detached_mode_vault_path(prop_map, resource_hostname, resource_name);
+
+        if (in_host_list) {
+            rodsLong_t resc_id = 0;
+
+            ret = resc_mgr.hier_to_leaf_id(resource_name, resc_id);
+            if (!ret.ok()) {
+                logger::resource::warn("[resource_name={}] Detached mode failed to set RESOURCE_HOST to {}."
+                                       "  Failed to call hier_to_leaf_id.",
+                                       resource_name.c_str(),
+                                       local_hostname);
+                return SUCCESS();
+            }
+
+            rodsServerHost_t* resource_host = nullptr;
+            ret = irods::get_resource_property<rodsServerHost_t*&>(resc_id, irods::RESOURCE_HOST, resource_host);
+
+            if (!ret.ok() || !resource_host) {
+                logger::resource::warn("[resource_name={}] Detached mode failed to set RESOURCE_HOST to {}."
+                                       "  Failed to get resource property irods::RESOURCE_HOST.",
+                                       resource_name.c_str(),
+                                       local_hostname);
+                return SUCCESS();
+            }
+
+            if (!resource_host) {
+                logger::resource::warn("[resource_name={}] Detached mode failed to set RESOURCE_HOST to {}."
+                                       "  The resource_host from resource property irods::RESOURCE_HOST was null.",
+                                       resource_name.c_str(),
+                                       local_hostname);
+                return SUCCESS();
+            }
```

Comment:
This is dead code. We should either remove this or remove the `nullptr` check on 160.